packages/compass-e2e-tests/smoke-test.ts

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/env npx ts-node
+import { createWriteStream, existsSync, promises as fs } from 'fs';
+import path from 'path';
+import yargs from 'yargs';
+import type { Argv } from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import https from 'https';
+import { pick } from 'lodash';
+import { handler as writeBuildInfo } from 'hadron-build/commands/info';
+
+const argv = yargs(hideBin(process.argv))
+  .scriptName('smoke-tests')
+  .detectLocale(false)
+  .version(false)
+  .strict()
+  .option('bucketName', {
+    type: 'string',
+    default: () => process.env.EVERGREEN_BUCKET_NAME,
+  })
+  .option('bucketKeyPrefix', {
+    type: 'string',
+    default: () => process.env.EVERGREEN_BUCKET_KEY_PREFIX,
+  })
+  .option('devVersion', {
+    type: 'string',
+    // For dev versions we need this from evergreen. For beta or stable (or by
+    // default, ie. when testing a locally packaged app) we get it from the
+    // package.json
+    default: () => process.env.DEV_VERSION_IDENTIFIER,
+  })
+  .option('isWindows', {
+    type: 'boolean',
+    default: () => process.env.IS_WINDOWS === 'true',
+  })
+  .option('isOSX', {
+    type: 'boolean',
+    default: () => process.env.IS_OSX === 'true',
+  })
+  .option('isRHEL', {
+    type: 'boolean',
+    default: () => process.env.IS_RHEL === 'true',
+  })
+  .option('isUbuntu', {
+    type: 'boolean',
+    default: () => process.env.IS_UBUNTU === 'true',
+  })
+  .option('arch', {
+    type: 'string',
+    choices: ['x64', 'arm64'],
+    demandOption: true,
+    default: () => {
+      if (process.env.ARCH) {
+        return process.env.ARCH;
+      }
+      return process.arch;
+    },
+  })
+  .option('skipDownload', {
+    type: 'boolean',
+    description: "Don't download all assets before starting",
+  })
+  .option('extension', {
+    type: 'string',
+    description:
+      'If specified it will only run the smoke tests for the specified installer/package',
+  })
+  .check((argv) => {
+    if (!argv.skipDownload) {
+      if (!(argv.bucketName && argv.bucketKeyPrefix)) {
+        throw new Error(
+          'Either supply EVERGREEN_BUCKET_NAME and EVERGREEN_BUCKET_KEY_PREFIX or specify --skip-download'
+        );
+      }
+    }
+
+    if (!(argv.isWindows || argv.isOSX || argv.isUbuntu || argv.isRHEL)) {
```

Comment:
We could detect these as a default, but I'm not sure we want to duplicate [the logic](https://github.com/mongodb-js/compass/blob/c8f0a6b3869fba35ab8ad80e2f3095e2eab09224/.evergreen/print-compass-env.sh#L7-L28)?

I spent way too much time thinking about this and going back and forth.

Reply:
Should we also check to ensure these are mutually exclusive?

Reply:
I've been thinking about that. Either make them mutually exclusive or allow you to "OR" them together. Then you can use this to download everything if you want to or at least test multiple .zip files at a time. Sure it will break later, but much of the code could run on mac for testing.